pkg/middleware/request_metrics.go

```diff
@@ -40,7 +41,7 @@ func init() {
 			Help:      "Histogram of latencies for HTTP requests.",
 			Buckets:   defBuckets,
 		},
-		[]string{"handler", "status_code", "method"},
+		[]string{"handler", "status_code", "method", "errorSource"},
```

Comment:
```suggestion
		[]string{"handler", "status_code", "method", "error_source"},
```

Reply:
👍 

I also don't like the name in general but it's the best I've come up with so far. Perhaps `status_source` is better?

Reply:
if it's populated for all statuses, maybe `status_source` makes sense...

Reply:
That is to say, it's _only_ populated for errors. Should we populate this all the time? Or does that even make sense?

Reply:
`response_source` might be another alternative....

`status_source` feels like a good name when there is not an error. 

`error_source` feels better when the error comes from downtream services. 

🤔